app/controllers/ProductController.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Controllers;
+
+class ProductController
+{
+    public function show($id): void
+    {
+
+        $product = $this->getProductById($id);
+
+        if ($product)
```

Comment:
Should be wrapped by Response instance, that will handle everything.

ex.:
$this->response->json($data, 200);
$this->response->json(['message' => 'Not Found'], 404);

// future update:
$this->response->view('view.php', $data, $status);

Reply:
Changed